### ISSUE TITLE
[no-unsafe-*] no-unsafe rules not compatible with WebDriver.io (custom entry in types array for compiler options)

### ISSUE THREAD

**Issue by ngregory-rbi** [typescript-eslint/typescript-eslint#4839]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unsafe-call": ["error"],
    "@typescript-eslint/no-unsafe-assignment": ["warn"],
    "@typescript-eslint/no-unsafe-member-access": ["error"],
  }
}
```

```TS
// your repro code case
import type { RemoteOptions } from 'webdriverio';

import { remote } from 'webdriverio';

const options: RemoteOptions = { capabilities: {} };

async function test() {
  const browser = await remote(options);
  browser.url('');
}
```

```json
{
  "compilerOptions": {
    "types": ["webdriverio/async"],
  }
}
```

**Expected Result**

I expect no errors.

**Actual Result**

`no-unsafe-call` highlights both the `remote` call and the `url` call.

`no-unsafe-assignment` highlights the assignment of the `browser` variable.

`no-unsafe-member-access` highlights the `url` method access.

**Additional Info**

WebDriver.IO, for some reason, requires a line to be added to the `tsconfig.json` compiler options (specifically an element for the `types` array). This is unusual, as most packages can have types picked up automatically. The `@typescript-eslint/parser` package may not be able to pick these up.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.17.0` |
| `@typescript-eslint/parser`        | `5.17.0` |
| `TypeScript`                       | `4.6.3` |
| `ESLint`                           | `1.2.1` |
| `node`                             | `14.17.3` |
|`webdriverio` | `7.19.5` |

**Comment by bradzacher** [comment]
Could you please provide a repro repo for this?

**Comment by ngregory-rbi** [comment]
My apologies for replying late on this. I've been pretty busy with work. I'll hopefully get around to this by this week.

**Comment by ngregory-rbi** [comment]
I'm going to close this issue for now, as I have been too busy with my work. Although, I would not be surprised if this was simply an issue of having to restart the ESLint server. I'll bring this issue back if and/or when I have the proper bandwidth to reproduce it.
